influxql/ast.go

```diff
@@ -642,6 +642,36 @@ type SelectStatement struct {
 	FillValue interface{}
 }
 
+// HasDerivative returns true if one of the field in the statement is a
+// derivative aggregate
+func (s *SelectStatement) HasDerivative() bool {
```

Comment:
Will we need to do the same for other function calls?  If so, then maybe `HasCall(name string) bool` or `CallsFunction(name string) bool`.